### ISSUE TITLE
How to uninstall microg services?

### ISSUE THREAD

**Issue by mscofield0** [microg/GmsCore#1477]
**Describe the bug**

I installed UnifiedNlp Gapps and quickly realized I don't need that one, but the no gapps version. I uninstalled UnifiedNlp but it didn't uninstall the service apps... The reason I installed gapps instead of no gapps is because fdroid doesn't have a non-legacy no gapps in the repo. If I now try to install the non gapps version, it detects I already have a unifiednlp installed and the installation fails. 

**To Reproduce**

Install UnifiedNlp from fdroid and uninstall it.

**Expected behavior**

I want to uninstall the damn service apps

**System**
Android Version: 11
Custom ROM: Samsung

**Comment by fynngodau** [comment]
Can you not uninstall the application you installed from the system settings?

**Comment by mscofield0** [comment]
It's not in the system apps list.

**Comment by fynngodau** [comment]
What happens if you launch this application from F-Droid?

**Comment by mscofield0** [comment]
I can't run the app, it's a service.

**Comment by fynngodau** [comment]
Okay, so with "service apps", which applications are you referring to, exactly? Could you give me their package name? Could you try to uninstall them with `adb uninstall`?

**Comment by mscofield0** [comment]
I don't know the package name, I can't even search for it in adb because of that.

Also can a Samsung phone with Google preinstalled even work with no gapps? Is no gapps exclusive for custom roms?

**Comment by fynngodau** [comment]
That will not work at all, and that is your problem: you are only able to use and install the "no gapps" version if you have no Google apps on your device. See the description [on the F-Droid listing](https://f-droid.org/en/packages/com.google.android.gms/):

> If this app shows as installed or as updateable but you don’t
recall installing it, you should completely ignore it as it will be useless to
you and you will not be able to uninstall it nor update it. This is because the
Google version comes preinstalled in your device already.

To recap, the application is shown as installed, but what is actually installed on your device is Google Play services. The "no gapps" version has the same package name as Google Play services, which is why F-Droid cannot distinguish the two.

**Comment by mscofield0** [comment]
Ah, okay, but I still want to remove the services. 

Also I'm having a hard time making it work, it doesn't bind to the system even after installing the [xposed](https://repo.xposed.info/module/de.r3w6.xposedunifiednlp) module. And restarting doesn't help.

**Comment by fynngodau** [comment]
Sorry, I don't know how to get UnifiedNlp working. It has not been updated in a while, so I'm not sure whether it is working with modern systems.

If you want to remove Google Play services I think the easiest way is to install a custom ROM. If you would like to try out microG instead of Google Play services, it would be best to use https://lineage.microg.org in case your device is supported by Lineage. If you want to keep your current ROM, provided that root is available on your device, you could try out the app [/d/gapps](https://f-droid.org/en/packages/org.droidtr.deletegapps/), though I haven't tried it.

**Comment by mscofield0** [comment]
I don't want to uninstall Google Play services, I want to uninstall the microg services that were installed alongside UnifiedNlp

**Comment by fynngodau** [comment]
If you didn't install any additional applications and uninstalled the gapps version of UnifiedNlp which you had initially installed, then no app that is part of microG is left on your device.

Even though F-Droid shows UnifiedNlp (NO GAPPS) as installed, which is not true.

**Comment by mscofield0** [comment]
I don't believe that because look at what its showing me when I try to install the apk.

![image](https://user-images.githubusercontent.com/69715407/119188367-da872b00-ba7a-11eb-81f4-b98ca02c87ff.png)

**Comment by fynngodau** [comment]
Is this the "no gapps" version?

**Comment by mscofield0** [comment]
Yep

**Comment by fynngodau** [comment]
It is *impossible* to have both "UnifiedNlp (NO GAPPS)" and Google Play services installed on the same device at the same time.

Both devices have the same package name, though. This is why the Android system thinks that your APK can be installed as an update to Google Play services, and this is the reason the message is phrased like this.

But then, installation fails, because UnifiedNlp is actually a completely different app, by a completely different developer, signed with a different key.

So, UnifiedNlp is *not* installed on your system, unless you installed the "UnifiedNlp (GAPPS)" version again in the meantime.

**Comment by mscofield0** [comment]
Oh lol I'm dumb, you've literally just said that above and I didn't process it. 

Well, time to ask why UnifiedNlp doesn't work in their repo. 

Thanks for sitting down and guiding me through this.

**Comment by fynngodau** [comment]
Cheers.
